SaltSquareIO/sports-app-react-fe
# Validation and confirm password

[CLICKUP_TICKET_ID](https://app.clickup.com/t/86c0hdf8z)

## Summary of change

This PR implements the following features and improvements for the registration form:

1. Added indicators to distinguish required fields (email and password) from optional fields. The "Register" button remains disabled until all required fields are input.
2. Added validation logic for the input fields:
  - Email must follow the format `x@x.x`.
  - Password must have between 6 and 100 characters including at least one uppercase letter, one digit and one special character. 
  - If validation fails upon clicking "Register", an error message will be displayed and no values will be printed out in the console. 
3. Added a confirm password field to ensure the password is input correctly. Both password fields are required and follow the same validation. If the passwords do not match, an error message will be shown and no values will be printed out in the console. 

## QA Notes

To test this PR, follow these instructions:

1. Required fields:
  - Navigate to the registration page and check that the email and password fields are required.
  - Ensure that the "Register" button is disabled until these fields are input.
 
2. Email Validation:
 - Input incorrect email formats (e.g. missing `@` or domain) and verify that the error message "Invalid email address format." is displayed. 
 - Input a valid email and ensure no error is displayed. 

3. Password Validation:
 - Test the password field with inputs that:
   - Are too short (less than 6 characters).
   - Do not contain at least one uppercase letter, number and special character. 
 - Confirm that the error message is displayed until all requirements are satisfied. 

4. Confirm Password:
 - Enter a password in the "Password" field, then a different one in the "Confirm Password" field and verify that the error message "Passwords do not match." appears. 
 - Ensure that the form cannot be submitted unless both password fields match. 

5. Submit Button behavior:
 - Verify that the "Register" button is disabled until all required fields are input. 
 - Ensure that no values are printed out in the console if any validation fails.

## Screenshots

### Required fields and Submit Button behavior

![Required fields and Submit Button behavior](https://github.com/user-attachments/assets/ee2d4b14-8a39-4eec-acc8-a7f745ed6e90)

### Email Validation

![Email Validation](https://github.com/user-attachments/assets/c5cc6a8a-c6bb-4429-a8c0-bb74a172a5f0)

### Password Validation

![Password Validation](https://github.com/user-attachments/assets/05588243-dc23-4a49-9311-90aeaff65dd0)

### Confirm Password

![Confirm Password](https://github.com/user-attachments/assets/03436689-a571-4ad9-a966-ccd64ecef039)
